pkg/server/config.go

```diff
@@ -42,14 +45,12 @@ type Config struct {
 	IssuerAllowlist   []string
 }
 
+const (
+	SourceSystemAuthConfigRegex = `gha\:\/\/(\d+)\?(private_key|kms_id)=([\s\S]*)`
```

Comment:
I'm not sure we want to use a regex here versus `url.Parse` and extracting what we care about. `url.Parse("gha://app-id?private_key=babdf")` will parse and you can check the Protocol and iterate over the URL Params nicely.

Reply:
`url.Parse` doesn't love the fact that that private_key parameter is an actual PEM encoded PK. I tried multiple ways of escaping that string and it throws errors like `net/url: invalid control character in URL` no matter what I do. 

I've got a few more things to try but this doesn't look like it will work.

Reply:
right, you'd have to hex/b64 encode the string for it to be urlsafe (and use urlsafe b64).